From 2471034e8bb7db1fb0f002019d452ca921c87527 Mon Sep 17 00:00:00 2001
From: Ryota MIBU <r-mibu@cq.jp.nec.com>
Date: Sat, 2 Jun 2012 17:39:31 +0900
Subject: [PATCH] fixed create_filter in config.cgi

---
 sliceable_switch/config.cgi |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sliceable_switch/config.cgi b/sliceable_switch/config.cgi
index e2fe9b1..03c07d3 100755
--- a/sliceable_switch/config.cgi
+++ b/sliceable_switch/config.cgi
@@ -589,6 +589,10 @@ sub add_filters(){
     my $content_string = get_request_body();
     my $content = from_json($content_string);
 
+    if(ref($content) ne 'ARRAY'){
+        $content = [$content];
+    } 
+
     foreach my $filter (@{$content}){
 	my @criteria = ();
 	$criteria[0] = oct(${$filter}{'priority'});
-- 
1.7.9.5

