diff --git a/nova/network/quantum/quantum_connection.py b/nova/network/quantum/quantum_co
index ce07bc1..68107b4 100644
--- a/nova/network/quantum/quantum_connection.py
+++ b/nova/network/quantum/quantum_connection.py
@@ -117,6 +117,9 @@ class QuantumClientConnection(object):
                 port_id = p["id"]
                 port_get_resdict = self.client.show_port_attachment(net_id,
                                 port_id, tenant=tenant_id)
-                if attachment_id == port_get_resdict["attachment"]["id"]:
+                # For Quantum API v1.0
+                if attachment_id == port_get_resdict["attachment"].get("id"):
+                    return (net_id, port_id)
+                if attachment_id == port_get_resdict["attachment"]:
                     return (net_id, port_id)
         return (None, None)
