diff --git a/sliceable_switch/config.cgi b/sliceable_switch/config.cgi
index e2fe9b1..03c07d3 100755
--- a/sliceable_switch/config.cgi
+++ b/sliceable_switch/config.cgi
@@ -589,6 +589,10 @@ sub add_filters(){
     my $content_string = get_request_body();
     my $content = from_json($content_string);
 
+    if(ref($content) ne 'ARRAY'){
+        $content = [$content];
+    } 
+
     foreach my $filter (@{$content}){
 	my @criteria = ();
 	$criteria[0] = oct(${$filter}{'priority'});
diff --git a/sliceable_switch/filter.c b/sliceable_switch/filter.c
index 520778a..39ba7d9 100644
--- a/sliceable_switch/filter.c
+++ b/sliceable_switch/filter.c
@@ -28,7 +28,7 @@
 #include "filter.h"
 
 
-#define FILTER_DB_UPDATE_INTERVAL 30
+#define FILTER_DB_UPDATE_INTERVAL 2
 
 
 typedef struct {
diff --git a/sliceable_switch/slice.c b/sliceable_switch/slice.c
index 45d22c9..4a47752 100644
--- a/sliceable_switch/slice.c
+++ b/sliceable_switch/slice.c
@@ -30,7 +30,7 @@
 #include "filter.h"
 
 
-#define SLICE_DB_UPDATE_INTERVAL 30
+#define SLICE_DB_UPDATE_INTERVAL 2
 
 #define BINDING_AGING_INTERVAL 60
 #define BINDING_TIMEOUT 3600
