From 19f00d59e1c17870a327c275ade18f2bfddb3192 Mon Sep 17 00:00:00 2001
From: Ryota MIBU <r-mibu@cq.jp.nec.com>
Date: Sat, 2 Jun 2012 23:16:54 +0900
Subject: [PATCH 2/2] support http_proxy

---
 functions |    8 +++++++-
 stack.sh  |    2 +-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/functions b/functions
index 5114de1..52067ce 100644
--- a/functions
+++ b/functions
@@ -288,10 +288,16 @@ function pip_install {
     else
         CMD_PIP=/usr/bin/pip-python
     fi
+    local _PIP_PROXY
+    if [ -n "$http_proxy" ]; then
+      _PIP_PROXY="--proxy=$http_proxy"
+    elif [ -n "$https_proxy" ]; then
+      _PIP_PROXY="--proxy=$https_proxy"
+    fi
     sudo PIP_DOWNLOAD_CACHE=/var/cache/pip \
         HTTP_PROXY=$http_proxy \
         HTTPS_PROXY=$https_proxy \
-        $CMD_PIP install --use-mirrors $@
+        $CMD_PIP install --use-mirrors $_PIP_PROXY $@
 }
 
 
diff --git a/stack.sh b/stack.sh
index ef3a930..e68b9cb 100755
--- a/stack.sh
+++ b/stack.sh
@@ -1722,7 +1722,7 @@ if is_service_enabled g-reg; then
 
     ADMIN_USER=admin
     ADMIN_TENANT=admin
-    TOKEN=`curl -s -d  "{\"auth\":{\"passwordCredentials\": {\"username\": \"$ADMIN_USER\", \"password\": \"$ADMIN_PASSWORD\"}, \"tenantName\": \"$ADMIN_TENANT\"}}" -H "Content-type: application/json" http://$HOST_IP:5000/v2.0/tokens | python -c "import sys; import json; tok = json.loads(sys.stdin.read()); print tok['access']['token']['id'];"`
+    TOKEN=`http_proxy= curl -s -d  "{\"auth\":{\"passwordCredentials\": {\"username\": \"$ADMIN_USER\", \"password\": \"$ADMIN_PASSWORD\"}, \"tenantName\": \"$ADMIN_TENANT\"}}" -H "Content-type: application/json" http://$HOST_IP:5000/v2.0/tokens | python -c "import sys; import json; tok = json.loads(sys.stdin.read()); print tok['access']['token']['id'];"`
 
     # Option to upload legacy ami-tty, which works with xenserver
     if [[ -n "$UPLOAD_LEGACY_TTY" ]]; then
-- 
1.7.9.5

