From f08b2f864a73b993b77dcedc5bc3b19df73dfee8 Mon Sep 17 00:00:00 2001
From: Akihiro MOTOKI <motoki@da.jp.nec.com>
Date: Mon, 25 Jun 2012 14:04:10 +0900
Subject: [PATCH] Multi node support with Quantum.

---
 stack.sh |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/stack.sh b/stack.sh
index e68b9cb..1bc3420 100755
--- a/stack.sh
+++ b/stack.sh
@@ -626,7 +626,7 @@ if is_service_enabled swift; then
     # storage service
     git_clone $SWIFT_REPO $SWIFT_DIR $SWIFT_BRANCH
 fi
-if is_service_enabled g-api n-api; then
+if is_service_enabled g-api n-api n-cpu; then
     # image catalog service
     git_clone $GLANCE_REPO $GLANCE_DIR $GLANCE_BRANCH
 fi
@@ -641,7 +641,7 @@ fi
 if is_service_enabled quantum; then
     git_clone $QUANTUM_CLIENT_REPO $QUANTUM_CLIENT_DIR $QUANTUM_CLIENT_BRANCH
 fi
-if is_service_enabled q-svc; then
+if is_service_enabled quantum; then
     # quantum
     git_clone $QUANTUM_REPO $QUANTUM_DIR $QUANTUM_BRANCH
     if [[ "$Q_PLUGIN" = "nec" ]]; then
@@ -672,7 +672,7 @@ fi
 if is_service_enabled swift; then
     cd $SWIFT_DIR; sudo python setup.py develop
 fi
-if is_service_enabled g-api n-api; then
+if is_service_enabled g-api n-api n-cpu; then
     cd $GLANCE_DIR; sudo python setup.py develop
 fi
 cd $NOVA_DIR; sudo python setup.py develop
@@ -682,7 +682,7 @@ fi
 if is_service_enabled quantum; then
     cd $QUANTUM_CLIENT_DIR; sudo python setup.py develop
 fi
-if is_service_enabled q-svc; then
+if is_service_enabled quantum; then
     cd $QUANTUM_DIR; sudo python setup.py develop
 fi
 if is_service_enabled m-svc; then
-- 
1.7.9.5

