diff --git a/stack.sh b/stack.sh
index d9e77e7..0ee2333 100755
--- a/stack.sh
+++ b/stack.sh
@@ -786,6 +786,8 @@ fi
 if is_service_enabled quantum; then
     # quantum
     git_clone $QUANTUM_REPO $QUANTUM_DIR $QUANTUM_BRANCH
+    # FIXME(r-mibu): get review code
+    git fetch https://review.openstack.org/openstack/quantum refs/changes/64/10664/1 && git cherry-pick FETCH_HEAD
 fi
 if is_service_enabled m-svc; then
     # melange
